Lib/tarfile.py

```diff
@@ -237,10 +238,12 @@ def copyfileobj(src, dst, length=None, exception=OSError, bufsize=None):
     """
     bufsize = bufsize or 16 * 1024
     if length == 0:
-        return
+        return 0
+
     if length is None:
+        start_pos = dst.tell()
         shutil.copyfileobj(src, dst, bufsize)
-        return
+        return dst.tell() - start_pos
```

Comment:
This is indeed used only by my code, from what I could gather calling `copyfileobj` with `length=None` is never done and this is not a documented interface.

If we inline this part of the code in `addbuffer` I think we should remove the support for `length=None` and raise an error here to avoid mistakes.